### DISCUSSION TITLE
mimic illustrator blob brush

### DISCUSSION THREAD

**Discussion by gritnz** [spakin/SimpInkScr#96] [Ideas]
Can a script be made to mimic illustrators blob brush? 

Step 1 with the 'calligraphy bush selected' in 'brush' mode, Select "????" this will turn on the script. Once the tablet pen or mouse lifts off the surface it triggers the script to "CTRL+A (select all) followed by "CTRL+K" (combine)

Step 2 once you have finished using the 'brush' as a 'blob brush' de-select "????" to go back to normal brush mode

I know you can do this manually but it is a real pain to do continually for hours on end.

**Comment by spakin** [comment]
Alas, I don't know of a good programmatic match to Illustrator's blob brush.  The following is the best approximation I managed to produce:

```Python
# Prepare the canvas.
canvas.true_width = 512
canvas.true_height = 768
canvas.viewbox = [0.0, 0.0, 512, 768]

# Define a brush with a rectangular cross section.
cross_section = path(
    [
        Move(15, 25),
        Line(60, 8),
        Line(65, 24),
        Line(20, 40),
        ZoneClose(),
    ],
    stroke='none',
    fill='none')
blob_brush = path_effect('skeletal',
                         copytype='repeated_stretched',
                         pattern=str(cross_section)[4:-1],
                         prop_units=True,
                         spacing=-2,
                         vertical_pattern=True)

# Draw a ribbon using the brush defined above.
ribbon = path(
    [
        Move(448, 704),
        Curve(320, 512, 126, 66, 256, 64),
        Curve(384, 62, 194, 513, 64, 704),
    ],
    stroke='none',
    fill='hotpink')
ribbon.apply_path_effect(blob_brush)
```
<img src="https://github.com/spakin/SimpInkScr/assets/650041/01c6c09a-4bb7-4d1d-bfb1-ca136115212d" width="128" height="192">

That code uses the _Pattern Along Path_ live path effect to draw a path with a shaped pen.

As I understand Illustrator's blob brush, the main difference is that a blob brush creates a single object while the above is an aggregate of many cross-section objects.  Hence, if you set a stroke, each cross section is stroked individually rather than stroking the object as a whole.
